ledger/db/src/mint_config_store.rs

```diff
@@ -0,0 +1,631 @@
+        set_mint_config_tx: &SetMintConfigTx,
+        db_transaction: &mut RwTransaction,
+    ) -> Result<(), Error> {
+        // All mint configurations must have the same token id.
+        if set_mint_config_tx
+            .prefix
+            .configs
+            .iter()
+            .any(|mint_config| mint_config.token_id != set_mint_config_tx.prefix.token_id)
+        {
+            return Err(Error::InvalidMintConfig(
+                "All mint configurations must have the same token id".to_string(),
+            ));
+        }
+
+        // MintConfigs -> ActiveMintConfigs
+        let active_mint_configs = ActiveMintConfigs::from(set_mint_config_tx);
+
+        // Store in database
+        db_transaction.put(
+            self.set_mint_config_tx_by_nonce,
+            &set_mint_config_tx.prefix.nonce,
+            &encode(set_mint_config_tx),
+            WriteFlags::NO_OVERWRITE, /* this ensures we do not overwrite a nonce that was
+                                       * already used */
+        )?;
+
+        db_transaction.put(
+            self.active_mint_configs_by_token_id,
+            &u32_to_key_bytes(set_mint_config_tx.prefix.token_id),
+            &encode(&active_mint_configs),
+            WriteFlags::empty(),
+        )?;
+
+        Ok(())
+    }
+
+    /// Get mint configurations for a given token.
+    pub fn get_active_mint_configs(
+        &self,
+        token_id: TokenId,
+        db_transaction: &impl Transaction,
+    ) -> Result<Vec<ActiveMintConfig>, Error> {
+        let token_id_bytes = u32_to_key_bytes(*token_id);
+        match db_transaction.get(self.active_mint_configs_by_token_id, &token_id_bytes) {
+            Ok(bytes) => {
+                let active_mint_configs: ActiveMintConfigs = decode(bytes)?;
+                Ok(active_mint_configs.configs)
+            }
+            Err(lmdb::Error::NotFound) => Ok(Vec::new()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    /// Update the total minted amount for a given MintConfig.
+    pub fn update_total_minted(
+        &self,
+        mint_config: &MintConfig,
+        amount: u64,
+        db_transaction: &mut RwTransaction,
+    ) -> Result<(), Error> {
+        if amount > mint_config.mint_limit {
+            return Err(Error::MintLimitExceeded(amount, mint_config.mint_limit));
+        }
+
+        // Get the active mint configs for the given token.
+        let mut active_mint_configs =
+            self.get_active_mint_configs(TokenId::from(mint_config.token_id), db_transaction)?;
+
+        // Find the active mint config that matches the mint config we were given.
+        let active_mint_config = active_mint_configs
+            .iter_mut()
+            .find(|active_mint_config| active_mint_config.mint_config == *mint_config)
+            .ok_or_else(|| Error::InvalidMintConfig("Mint config not found".to_string()))?;
+
+        // Update the total minted amount.
+        active_mint_config.total_minted = amount;
```

Comment:
Consider enforcing that `amount > total_minted` here, and adding a test that exercises it.